### ISSUE TITLE
Windows scripts path resolution broken

### ISSUE THREAD

**Issue by stonesbg** [evilmartians/lefthook#367]
### Summary

I was trying to setup a python commit-msg script for a windows machine and the path was resolving wrong.  Example output

```shell
Lefthook v1.2.0
RUNNING HOOK: commit-msg
C:\Users\bgarvey\AppData\Local\Programs\Python\Python310\python.exe: can't open file "C:\\dev\\pm\\apm-core\\'C:\\dev\\pm\\apm-core\\.lefthook\\commit-msg\\jira.py'": [Errno 22] Invalid argument
```

I have checked past releases and have gone back to v1.1.0 which doesn't seem to have the issue. I have also checked that it isn't just a python issue and created a node script and that too had issues trying to run saying it couldn't find module.

_NOTE: From the same directory in wsl there is no issues parsing the path and the pre-commit/commit-msg work perfectly_

### Steps to reproduce

- On windows install lefthook using npm
- Setup a hook for pre-commit that uses a script (node or python)
- Execute the hook by a git commit or by running `lefthook run pre-commit`

### Expected results

Expected that the pre-commit script will run on windows without issue

### Actual results

Fails to create the file path correctly seems to be doing a merge between the base repo directory and the full path to the file.

**Comment by mrexox** [comment]
Hey @stonesbg! Thank you for the issue. Could you provide your `lefthook.yml` file content?

**Comment by stonesbg** [comment]
Sure it is just this:

```yaml
```

**Comment by mrexox** [comment]
Well, looks like it is related to how `filepath.Join` works on Windows. Unfortunately, I can't debug this now, because I only have an access to Linux machine now. Do you have a will to dig into some Golang stuff? I believe the breaking change was in [this](https://github.com/evilmartians/lefthook/commit/ac4212247eecf31e796dbd169414b8342e2c7c57#diff-9d7b53114cec5ab660f7846b7de992bee92fe8bc820999503e4b27723658a9d9L74) commit. So, I'd try printing the values of `scriptDirs` in `run.go:100` and in `runner.go:78` to see where the duplicate join appears.

If you don't have an opportunity to dig into this, I can ask one of my colleagues to help :open_hands:

**Comment by stonesbg** [comment]
@mrexox I took a stab at a resolution https://github.com/evilmartians/lefthook/pull/369

**Comment by mrexox** [comment]
Hey! I've just released v1.2.1. Please, check it. The issue should be resolved there.
